packages/sdk/src/browser-wallets/xverse/index.ts

```diff
@@ -25,10 +49,64 @@ function isInstalled() {
  * @returns An array of WalletAddress objects.
  */
 async function getAddresses(
-  _network: BrowserWalletNetwork = "mainnet",
-  _readOnly?: boolean,
+  network: BrowserWalletNetwork = "mainnet",
+  readOnly?: boolean,
 ): Promise<WalletAddress[]> {
-  return [];
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("Xverse not installed");
+  }
+
+  if (readOnly) {
+    throw new OrditSDKError("Read only mode is not supported on Xverse");
+  }
+
+  const result: Array<{
+    publicKey: string;
+    address: string;
+    format: AddressFormat;
+  }> = [];
+
+  const handleOnFinish = (response: GetAddressResponse) => {
+    if (!response || !response.addresses || response.addresses.length !== 2) {
+      throw new OrditSDKError("Invalid address format");
+    }
+
+    response.addresses.forEach((addressObj) => {
+      const format = getAddressFormat(addressObj.address, network);
+
+      let fullPubKey = addressObj.publicKey;
+      if (format === "taproot") {
+        // For taproot addresses, Xverse returns the x-only public key.
+        fullPubKey = fromXOnlyToFullPubkey(addressObj.publicKey);
+      }
+
+      result.push({
+        publicKey: fullPubKey,
+        address: addressObj.address,
+        format,
+      });
+    });
+  };
+
+  const handleOnCancel = () => {
+    throw new BrowserWalletUserCancelledError("Request canceled by user.");
+  };
+
+  const xVerseOptions: GetAddressOptions = {
+    payload: {
+      purposes: ["ordinals", "payment"] as AddressPurpose[],
+      message: "Provide access to payment Address and Ordinals address", // Message is hardcoded for now
```

Comment:
this comment can be removed (for now = forever tbh)

Reply:
I think I left this year in case we want to change it in future, but yeah I think it is redundant LOL

Reply:
Removed!